CDP4Dal/Session.cs

```diff
@@ -729,25 +750,58 @@ public async Task Reload()
         /// <summary>
         /// Can a Cancel action be executed?
         /// </summary>
-        /// <returns>True is Cancel is allowed, otherwise false.</returns>
-        public bool CanCancel()
+        /// <param name="cancellationTokenSource">
+        /// Cancellation token source that will be cancelled <see cref="CancellationTokenSource"/>
+        /// </param>
+        /// <returns>
+        /// True if Cancel is allowed, otherwise false.
+        /// </returns>
+        private static bool CanCancel(CancellationTokenSource cancellationTokenSource)
         {
-            if (this.cancellationTokenSource == null)
+            if (cancellationTokenSource == null)
             {
                 return false;
             }
 
-            return this.cancellationTokenSource.Token.CanBeCanceled && !this.cancellationTokenSource.IsCancellationRequested;
+            return cancellationTokenSource.Token.CanBeCanceled && !cancellationTokenSource.IsCancellationRequested;
+        }
+
+        /// <summary>
+        /// Can a Cancel action be executed?
+        /// </summary>
+        /// <returns>
+        /// True if Cancel is allowed for at least on token, otherwise false.
+        /// </returns>
+        public bool CanCancel()
+        {
+            foreach (var cancellationTokenSourceKey in this.cancellationTokenSourceDictionary.Keys)
+            {
+                this.cancellationTokenSourceDictionary.TryGetValue(cancellationTokenSourceKey, out var cancellationTokenSource);
+
+                if (!CanCancel(cancellationTokenSource))
+                {
+                    continue;
+                }
+
+                cancellationTokenSource?.Cancel();
+
+                return true;
+            }
+
+            return false;
         }
 
         /// <summary>
         /// Cancel any Read or Open operation.

```

Comment:
you've also updated the `Update` method so the docs should state:

```
Cancel all running Read, Open and Write operations that are currently running
```
you can add a remark:

```
A Write Operation can be cancelled which will result in the Session will no longer waiting for the result. The operations are already dispatched to the E-TM-10-25 data source and will continue to run and complete there.
```